src/chmod/chmod.rs

```diff
@@ -40,7 +42,17 @@ pub fn uumain(args: Vec<String>) -> i32 {
     opts.optflag("R", "recursive", "change files and directories recursively");
     opts.optflag("h", "help", "display this help and exit");
     opts.optflag("V", "version", "output version information and exit");
-    // TODO: sanitize input for - at beginning (e.g. chmod -x testfile).  Solution is to add a to -x, making a-x
+    // sanitize input for - at beginning (e.g. chmod -x testfile). Remove
+    // the option and save it for later, after parsing is finished.
+    let re = Regex::new("^-{1}([rwx]|[0-7])").unwrap();
```

Comment:
Not really, it was already in place (maybe a leftover from a previous cleanup) and the resulting code is a bit easier to read. Without it, this is the best I can come up with my limited Rust-fu:

```rust
    for i in 0..args.len() {
        if let Some(first) = args[i].chars().nth(0) {
            if first != '-' {
                continue;
            }

            if let Some(second) = args[i].chars().nth(1) {
                match second {
                    'r' | 'w' | 'x' | 'X' | 's' | 't' | 'u' | 'g' | 'o' | '0' ... '7' => {
                        negative_option = Some(args.remove(i));
                        break;
                    },
                    _ => {}
                }
            }
        }
   }
```
What do you think?